### PR TITLE
Add own data to CDL dataset.

### PR DESCRIPTION
**Pull request by nilsleh**
As requested in #415 , this PR adds a data.py and new fake data.

### CODE REVIEW

**Comment by calebrob6** [comment]
Hmmm, I think we should add an explicit test for the cmap to get around this test coverage thing

**Comment by adamjstewart** [comment]
> Update: still not quite sure why test coverage would drop, but #430 should fix this.

Coverage is dropping because the new fake data does not have a cmap. We should add this to the data. Also, it seems like the actual `data.py` is missing from this PR.

**Review by calebrob6 — APPROVED** [review]

File: torchgeo/datasets/cdl.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ class CDL(RasterDataset):
         (2008, "e6aa3967e379b98fd30c26abe9696053"),
     ]
 
+    cmap: Dict[int, Tuple[int, int, int, int]] = {}
```
Comment (adamjstewart) [inline torchgeo/datasets/cdl.py:61]: This is already defined at the `RasterDataset` level, I don't think we need it here too

File: tests/datasets/test_cdl.py [hunk 2]
```diff
@@ -31,9 +31,21 @@ def dataset(
         monkeypatch.setattr(  # type: ignore[attr-defined]
             torchgeo.datasets.cdl, "download_url", download_url
         )
+        cmap = {
```
Comment (adamjstewart) [inline tests/datasets/test_cdl.py:34]: This cmap should be added directly to the file so we can test that TorchGeo can read the cmap from a file. See https://github.com/microsoft/torchgeo/tree/main/tests/data#raster-data for the code to add this to the file.

File: tests/data/cdl/data.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    Z = np.random.randint(size=(SIZE, SIZE), low=0, high=8)
+
+    src = rasterio.open(path, "w", **profile)
+    for i in range(1, profile["count"] + 1):
+        src.write(Z, i)
+
+
+directories = ["2020_30m_cdls", "2021_30m_cdls"]
+raster_extensions = [[".tif", ".tif.ovr"], [".tif", ".tif.ovr"]]
```
Comment (adamjstewart) [inline tests/data/cdl/data.py:40]: Can probably simplify this to a single list, no need to duplicate it

**Review by adamjstewart — COMMENTED** [review]
Few more things that can be improved in a follow-up PR

**Pull request merged**